### PR TITLE
8302685: Some javac unit tests aren't reliably closing open files

### PR DESCRIPTION
**Pull request by archiecobbs**
Some javac unit tests aren't reliably closing open files. Many were written prior to (or without utilizing) try-with-resources.

Leaving files and other resources open can cause problems during automated builds, etc.

This patch updates these tests to use try-with-resource.

Side note: there is at least one unit test that intentionally leaves a file open - `diags/examples/ProcUnclosedTypeFiles/processors/AnnoProc.java`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302685](https://bugs.openjdk.org/browse/JDK-8302685): Some javac unit tests aren't reliably closing open files

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to [e61e5714](https://git.openjdk.org/jdk/pull/12609/files/e61e57140e01da98831f2695f4145e88be6720c5)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [e61e5714](https://git.openjdk.org/jdk/pull/12609/files/e61e57140e01da98831f2695f4145e88be6720c5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12609/head:pull/12609` \
`$ git checkout pull/12609`

Update a local copy of the PR: \
`$ git checkout pull/12609` \
`$ git pull https://git.openjdk.org/jdk pull/12609/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12609`

View PR using the GUI difftool: \
`$ git pr show -t 12609`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12609.diff">https://git.openjdk.org/jdk/pull/12609.diff</a>

</details>

### CODE REVIEW

**Review by jddarcy — APPROVED** [review]
Looks fine; however, another engineer who works in the langtools area should look over the changes too.

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Comment by archiecobbs** [comment]
/integrate

Thanks for the reviews.

File: test/langtools/tools/javac/T8071847/T8071847.java [hunk 1]
```diff
@@ -83,9 +83,9 @@ byte[] hexToByte(String str) {
 
     File writeHexFile(String classFileName, String hexString) throws IOException {
         File f = new File(classFileName);
-        FileOutputStream output  = new FileOutputStream(f);
-        output.write(hexToByte(hexString));
-        output.close();
+        try (FileOutputStream output  = new FileOutputStream(f)) {
```
Comment (turbanoff) [inline test/langtools/tools/javac/T8071847/T8071847.java:86]:
```suggestion
        try (FileOutputStream output = new FileOutputStream(f)) {
```

Comment (archiecobbs) [reply test/langtools/tools/javac/T8071847/T8071847.java]: Thanks - should be fixed in 40a31c3bc13.

**Pull request closed without merging**